spec/session_tracker_spec.rb

```diff
@@ -2,6 +2,14 @@
 require 'webrick'
 require 'spec_helper'
 require 'json'
+require 'concurrent'
+
+# Enable reset of session_counts between each example
+module Bugsnag
+  class SessionTracker
+    attr_accessor :session_counts
```

Comment:
I think you can avoid opening/monkey-patching this class by having a `before` block call `Bugsnag.session_tracker.send_sessions` to flush `session_counts`, although you will probably need to set the delivery method to synchronous and clear `queue` right after the `send_sessions` call.

Reply:
To be honest, as long as the monkey-patching won't interfere with the tests I don't have a problem with it.

Reply:
A risk of monkey patching like this is that someone might introduce app code that uses the extra behavior (`Bugsnag::SessionTracker#session_counts=` in this case) and rely on the unit tests to make sure things work, which will be not as accurate as the tests introduce an extra method while the app code doesn't have this extra method.

Alternatively, you can do something similar to the global `before(:each)` block in spec_helper.rb and have `Bugsnag.instance_variable_set(:@session_tracker, Bugsnag::SessionTracker.new)` in a `before` block in this file (which has the benefit that anything in `Bugsnag.session_tracker` is reset if more instance variables are added to `Bugsnag::SessionTracker` later).

Reply:
That makes sense, and I quite like that last suggestion, more so than relying on `send_sessions`.

Reply:
Shouldn't you remove this monkey patch and the `require 'concurrent'` now that you're using `Bugsnag.instance_variable_set(:@session_tracker, Bugsnag::SessionTracker.new)`?

Also, I would put it in a `before(:each)` instead, then have an `after(:all)` so you have something fresh before the first test in this file also.